.github/workflows/cd-dev-api.yml

```diff
@@ -1,57 +1,50 @@
 on:
     push:
         branches:
-            - 'master'
-        paths:
-            - 'api/**'
-            - 'Cargo.lock'
-            - '**/cd-dev-api.yml'
+            - "master"
+        paths-ignore:
+            - "**/*.md"
 
 jobs:
-    build:
+    deploy:
         runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                service: ["api", "book", "borrow", "identity", "notification"]
+            fail-fast: false
+
+        env:
+            REGISTRY: ${{ secrets.AWS_ECR_REGISTRY }}
+            REPOSITORY: code-library-${{ matrix.service }}
+
         steps:
             - uses: actions/checkout@v2
-            - name: 'Build and save Docker image'
-              env:
-                  REGISTRY: ${{ secrets.AWS_ECR_REGISTRY }}
-                  REPOSITORY: code-library-api
-                  IMAGE_TAG: ${{ github.sha }}
+
+            - name: Check if service changed
+              id: did_change
               run: |
-                  docker build -t "$REGISTRY/$REPOSITORY:$IMAGE_TAG" -f api/Dockerfile .
-                  docker image save -o image "$REGISTRY/$REPOSITORY:$IMAGE_TAG"
-            - name: 'Upload the Docker image'
-              uses: actions/upload-artifact@v2
-              with:
-                  name: image
-                  path: image
-    deploy:
-        needs: build
-        runs-on: ubuntu-latest
-        steps:
+                  echo "::set-output name=did_change::$( \
+                  python3 tools/check_if_pr_changed_files.py \
+                  '${{ toJSON(github.event.commits.*.id) }}' \
+                  '["${{ matrix.service }}", "Cargo.lock", ".github/workflows/cd.yml"]' \
+                  )"
```

Comment:
Can you syntactically improve this of put some comments there?

Reply:
I though about moving that to a custom action, so it would  look something like:
```yml
- use: urhengulas/detect-file-changes
  with:
      files_to_watch:
          - ${{ matrix.service }}
          - Cargo.lock
          - .github/workflows/cd.yml
```
What would you think about that?